scripts/build.mjs

```diff
@@ -122,30 +122,59 @@ const createIndexes = async (results) => {
   )));
 };
 
-const addLearningObjectives = async () => {
-  const dir = path.join(__dirname, '../learning-objectives')
-  const [tree, es, pt] = await Promise.all([
-    loadYaml(path.join(dir, 'data.yml')),
-    loadYaml(path.join(dir, 'intl', 'es.yml')),
-    loadYaml(path.join(dir, 'intl', 'pt.yml')),
-  ]);
-
-  const flat = flattenLearningObjectives(tree);
-  const json = {
-    tree,
-    flat,
-    intl: { es, pt },
-    table: flat.map(key => ({
-      key,
-      es: es[key]?.title || es[key],
-      pt: pt[key]?.title || pt[key],
-    })),
-  };
-
-  await writeFile(
-    path.join(buildDir, 'learning-objectives.json'),
-    JSON.stringify(json, null, 2),
-  );
+const addLearningObjectives = async (validate) => {
+  
+  const parseObjectives = (validate) => new Promise((resolve) => {
+    const dir = path.join(__dirname, '../learning-objectives');
+    const dest = path.join(buildDir, 'learning-objectives.json')
+    const fd = openSync(validate ? '/dev/null' : dest, 'w');
+
+    const args = [
+      'curriculum-parser',
+      'learning-objectives',
+      dir,
+      // '--repo', repository,
+      // '--version', version,
+    ];
+
+    if (validate) {
+      args.push('--validate'); // TODO: put strict here?
+    }
+
+    console.log(args);
+    const child = spawn(
+      'npx',
+      args,
+      { stdio: [null, fd, 'pipe'] },
+    );
+  
+    const stderrChunks = [];
+    child.stderr.on('data', chunk => stderrChunks.push(chunk));
+
+    child.on('close', (code) => {
+      if (code > 0) {
+        const err = Object.assign(new Error(`Error parsing learning objectives`), {
+          stderr: stderrChunks.join(''),
+        });
+        console.error(err.stderr);
+        return resolve(err);
+      }
+  
+      console.log(`<= OK parsing objectives`);
+      return resolve({ dest });
+    });  
+  });
+  
+  const { dest } = await parseObjectives(validate);
+  console.log(buildDir, dest);
+  if (!validate) {
+    const json = await JSON.parse(await readFile(dest));
+    await writeFile(
+      path.join(buildDir, 'learning-objectives.json'),
+      JSON.stringify(json, null, 2),
+    );
```

Comment:
this step may be unnecessary since `fd` is the same (we are just reading and writing the same file)